mono/utils/mono-time.c

```diff
@@ -15,16 +15,23 @@
 #include <utils/mono-time.h>
 
 
-#define MTICKS_PER_SEC 10000000
+#define MTICKS_PER_SEC (10 * 1000 * 1000)
+
+gint64
+mono_msec_ticks (void)
+{
+	return mono_100ns_ticks () / 10 / 1000;
+}
 
 #ifdef HOST_WIN32
 #include <windows.h>
+#include <winbase.h>
 
-guint32
-mono_msec_ticks (void)
+gint64
+mono_msec_boottime (void)
 {
 	/* GetTickCount () is reportedly monotonic */
-	return GetTickCount ();
+	return GetTickCount64 ();
```

Comment:
it seems the cygwin/mingw build chokes on this:

```
mono-time.c: In function 'mono_msec_boottime':
mono-time.c:34:2: error: implicit declaration of function 'GetTickCount64' [-Werror=implicit-function-declaration]
mono-time.c:34:2: warning: nested extern declaration of 'GetTickCount64' [-Wnested-externs]
cc1: some warnings being treated as errors
```

which is strange given that I see -DWINVER=0x0600 so the API should be there...